### ISSUE TITLE
Toast Notifications are not accessible to screen readers

### ISSUE THREAD

**Issue by dougmacknz** [cultureamp/kaizen-design-system#341]
As per the current docs - toast notifications are created in response to a user action. This is mostly okay for sighted users who will hopefully see the toast pop up, but people using screen readers have no knowledge of the toast and won't be notified of its existence until (if) they navigate to the markup for the toast.

In terms of fixing this, ideally we could offer this accessibility out of the box with Kaizen. To do this we would wrap the toast markup inside an `aria-live` region (see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions). 
We would want to validate that this makes sense in every usage of the toast and won't have any negative consequences.

Additionally, we would need to make sure that usages of the Toast have it rendered on component mount, then toggled visibility via a prop on the component, because of this (as per Bootstrap docs):

> Note that the live region needs to be present in the markup before the toast is generated or updated. If you dynamically generate both at the same time and inject them into the page, they will generally not be announced by assistive technologies.

So we would need to change usages from:
```
{showError && <ToastNotification />}
```

to something like

```
<ToastNotification visible={showError} />
```

This is something that would need to be communicated to engineers. Without any knowledge of why they need to have it rendered on component mount, they will default to thinking it's better to not render it until required (I have seen this happening with modals as well).

Related: I also know of at least two cases (both written by me) where an Inline Notification is being used in response to a user action, like showing an error on submit of a form. In my case, this was required over a toast because the form is inside a modal. We might want to consider the same solution for those also.

Useful links:
https://getbootstrap.com/docs/4.3/components/toasts/#accessibility
https://medium.com/@sheribyrnehaber/designing-toast-messages-for-accessibility-fb610ac364be

**Comment by didoesdigital** [comment]
Useful link: https://www.scottohara.me/blog/2019/07/08/a-toast-to-a11y-toasts.html

**Comment by didoesdigital** [comment]
Some considerations:

- Our toasts usually come with actions as well such as "undo" or "learn more" links. How does a screen reader user navigate to such a thing? This might mean that announcing the info itself is not enough.
- We have some auto-hiding toasts. We might persistent them in the markup for screen reader users to have time to navigate to them as desired.
- In my experience, it's easier to manage a centralised announcer that juggles multiple sources of announcements than per component announcements because they can interrupt each other. Balancing this across our 2 front-end frameworks might get tricky.
- We might need to use a double announcer approach to minimise virtual DOM diffing/reconciliation confusing screen readers: https://almerosteyn.com/2017/09/aria-live-regions-in-react — I don't know of a package for Elm but there's one for [react-aria-live](https://www.npmjs.com/package/react-aria-live)

**Comment by makenosound** [comment]
FYI there’s some prior art in this space with https://github.com/reach/reach-ui/tree/master/packages/alert (and some of their other packages).
